components/script/dom/htmlinputelement.rs

```diff
@@ -895,6 +895,44 @@ impl HTMLInputElement {
                     textinput.set_content("#000000".into());
                 }
             }
+            atom!("time") => {
+                let mut textinput = self.textinput.borrow_mut();
+
+                let is_colon = |c: &str| c == ":";
+                let is_stop = |c: &str| c == ".";
+
+                let is_hour = |h: u8| h <= 23;
+                let is_minute = |m: u8| m <= 59;
+                let is_second = |s: u8| s <= 59;
+                let is_ms = |ms: u16| ms <= 999;
+
+                let is_valid = {
+                    let content = textinput.single_line_content().as_ref();
+                    let len = content.len();
+
+                    len >= 5 &&
+                        // HH:mm
+                        is_hour(content[0..2].parse().unwrap_or(99)) &&
+                        is_colon(&content[2..3]) &&
```

Comment:
`&content[2]` used to return a `char`, but that was removed b/c it doesn't work as expected when dealing with utf8. It would be fine for use as the only valid content is ASCII, but that's an edge case we're hitting here.
Maybe working with the underlying byte slice would work, but that's a dirty solution IMHO.